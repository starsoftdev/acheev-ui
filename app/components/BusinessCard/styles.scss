@import '~styles/settings/settings';

.businessCard {
  display: flex;
  height: 100%;
  width: 100%;
  flex-direction: column;
  text-align: center;
  padding-left: $rem-mx;
  padding-right: $rem-mx;
  margin-bottom: $rem-xl;
  @include breakpoint(small only) {
    margin-bottom: $rem-sm;
  }
  &__typeText {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    color: $white;
    font-size: $fs-tn;
    line-height: 23px;
    text-transform: uppercase;
    font-weight: bold;
  }
  &__availabilityStatus {
    width: 16px;
    height: 16px;
    border-radius: 50%;
    &--available {
      background-color: $success-color;
    }
    &--coming-soon {
      background-color: $warning-color;
    }
    &--out-of-stock, &--discontinued {
      background-color: $alert-color;
    }
  }
  &__image {
    display: inline-block;
    width: 78px;
    height: 78px;
    margin-bottom: $rem-sm;
    border-radius: 50%;
    box-shadow: 0 0 0 1px $light-gray;
    background-size: cover;
    background-position: center;
    position: relative;
  }
  &__title {
    margin-bottom: $rem-sm;
  }
  &__link {
    text-transform: none;
    text-decoration: none;
    color: #11b4e4;
  }
}

