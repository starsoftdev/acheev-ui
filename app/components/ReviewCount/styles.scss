@import '~styles/settings/settings';

.reviewCount {
  &__starSection {
    @include breakpoint(small only) {
      margin-bottom: $rem-sm;
    }
  }
  &__countSection {
    font-size: $fs-tn;
    color: $primary;
    min-width: 128px; // make products with different review count look consistent
  }
  &__star {
    justify-content: center;
  }
}
