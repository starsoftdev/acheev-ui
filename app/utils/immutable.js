// @flow

import { Iterable } from 'immutable';
import type { Map } from 'immutable';
import { get as _get } from 'lodash-es';

export const get = (
  collection: ?Map<*, *> | Object,
  path: string | number,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.get(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};

export const getIn = (
  collection: ?Map<*, *> | Object,
  path: Array<string | number>,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.getIn(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};
