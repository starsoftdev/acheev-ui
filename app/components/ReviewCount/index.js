// @flow

import * as React from 'react';
import cx from 'classnames';

import Link from 'components/Link';
import StarRating from 'components/StarRating';
import Icon from 'components/Icon';

import './styles.scss';

type Props = {
  reviewCount: number,
  ratingsAverage: number,
  className?: string,
  to?: string,
  prefixIcon?: string,
  type?: string,
  centered?: boolean,
};

const ReviewCount = ({
  reviewCount,
  ratingsAverage,
  className,
  to,
  prefixIcon,
  type,
  centered,
}: Props) => {
  const mergedClassName = cx('reviewCount row align-middle', className);
  const starsClassName = cx('reviewCount__starsContainer row align-middle', {
    'align-center': centered,
  });
  return (
    <div className={mergedClassName}>
      <div className="reviewCount__starSection column small-12 medium-shrink">
        <div className={starsClassName}>
          {prefixIcon && (
            <div className="shrink column npr">
              <Icon
                glyph={prefixIcon}
                size={20}
                className="reviewCount__icon"
              />
            </div>
          )}
          <div className="shrink column">
            <StarRating
              className="reviewCount__star"
              initialRating={ratingsAverage}
            />
          </div>
        </div>
      </div>
      <div className="reviewCount__countSection column small-12 medium-shrink">
        {ratingsAverage ? (
          <strong>({ratingsAverage})&nbsp;&nbsp;&nbsp;</strong>
        ) : (
          <strong>(0)&nbsp;&nbsp;&nbsp;</strong>
        )}
        <Link to={to}>
          {reviewCount} {type ? `${type} ` : ''}reviews
        </Link>
      </div>
    </div>
  );
};

export default ReviewCount;
