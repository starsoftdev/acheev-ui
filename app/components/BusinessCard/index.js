// @flow

import * as React from 'react';
import { history } from 'components/ConnectedRouter';
import Dotdotdot from 'react-dotdotdot';
import { get, getIn } from 'utils/immutable';

import CONFIG from '../../conf';
import Tooltip from 'components/Tooltip';
import ReviewCount from 'components/ReviewCount';
import Button from 'components/Button';
import Link from 'components/Link';

import './styles.scss';

type Props = {
  data: Object,
  hideReview?: boolean,
};

const BusinessCard = (props: Props) => {
  const { data, hideReview = false } = props;
  if (!data) return null;
  const type = get(data, '__t', '');
  const businessCategory = type === 'Doctor' ? 'Clinic' : type;
  const baseUrl =
    businessCategory === 'Dispensary'
      ? '/dispensaries'
      : `/${businessCategory.toLowerCase()}s`;
  const slug = get(data, 'slug');
  const changeRouteWithSlug = () => {
    history.push(`${baseUrl}/${slug}`);
  };
  const thumbnail =
    get(data, 'thumbnail') || getIn(data, ['business', 'thumbnail']);
  return (
    <div className="businessCard">
      <Link className="businessCard__link" to={`${baseUrl}/${slug}`}>
        {businessCategory ? (
          <Tooltip
            tooltipPosition="top"
            tooltipIndicator={false}
            tooltipContent={businessCategory}
            triggerFocus={false}
            triggerClick={false}
          >
            <div
              className="businessCard__image"
              style={{
                backgroundImage: `url('${CONFIG.APP.CDN_URL}/resize/78x78/${thumbnail}')`,
              }}
            />
          </Tooltip>
        ) : (
          <div
            className="businessCard__image"
            style={{
              backgroundImage: `url('${CONFIG.APP.CDN_URL}/resize/78x78/${thumbnail}')`,
            }}
          />
        )}
        <h4 className="businessCard__title">
          <Dotdotdot clamp={2}>{get(data, 'name')}</Dotdotdot>
        </h4>
      </Link>
      {!hideReview && (
        <ReviewCount
          className="align-center mb-mn"
          reviewCount={get(data, 'reviews').size}
          ratingsAverage={get(data, 'rating')}
          to={`${baseUrl}/${slug}/reviews`}
        />
      )}
      <div className="row">
        <div className="column text-center mt-md">
          <Button
            className="spacious secondary"
            onClick={() => changeRouteWithSlug()}
          >
            View
          </Button>
        </div>
      </div>
    </div>
  );
};

export default BusinessCard;
