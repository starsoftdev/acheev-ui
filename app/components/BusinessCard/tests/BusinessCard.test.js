import React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import { MemoryRouter } from 'react-router-dom';

import BusinessCard from 'components/BusinessCard';
import ReviewCount from 'components/ReviewCount';

const data = fromJS({
  name: 'Tweed',
  slug: 'tweed',
  thumbnail: 'https://lift.co/uploads/v6AIjoUMF7ozJw4UWzgF.png',
  __t: 'Producer',
  reviews: ['5980b8660952e830dce85e20', '5980b8660952e830dce85e0e'],
  rating: 4,
});

jest.mock('react-dotdotdot', () => 'div');
jest.mock('react-rating', () => 'div');

describe('<BusinessCard />', () => {
  it('should hide ReviewCount if hideReview={true}', () => {
    const renderedComponent = shallow(
      <MemoryRouter>
        <BusinessCard data={data} hideReview />
      </MemoryRouter>
    );
    expect(renderedComponent.contains(<ReviewCount />)).toEqual(false);
  });

  it("shouldn't throw error if data is not specified", () => {
    try {
      shallow(
        <MemoryRouter>
          <BusinessCard />
        </MemoryRouter>
      );
    } catch (error) {
      expect(error).toBeUndefined();
    }
  });

  it('renders data if provided', () => {
    const component = create(
      <MemoryRouter>
        <BusinessCard data={data} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
