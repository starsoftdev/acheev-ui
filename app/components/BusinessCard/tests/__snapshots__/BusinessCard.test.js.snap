// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BusinessCard /> renders data if provided 1`] = `
<div
  className="businessCard"
>
  <a
    className="businessCard__link"
    href="/producers/tweed"
    onClick={[Function]}
  >
    <div
      aria-haspopup={true}
      className="businessCard__image"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "backgroundImage": "url('https://images.lift.co/resize/78x78/https://lift.co/uploads/v6AIjoUMF7ozJw4UWzgF.png')",
        }
      }
    />
    <h4
      className="businessCard__title"
    >
      <div
        clamp={2}
      >
        Tweed
      </div>
    </h4>
  </a>
  <div
    className="reviewCount row align-middle align-center mb-mn"
  >
    <div
      className="reviewCount__starSection column small-12 medium-shrink"
    >
      <div
        className="reviewCount__starsContainer row align-middle"
      >
        <div
          className="shrink column"
        >
          <div
            className="starRating reviewCount__star"
          >
            <div
              emptySymbol={
                <Icon
                  glyph="IMAGE_MOCK"
                  size={16}
                />
              }
              fractions={2}
              fullSymbol={
                <Icon
                  glyph="IMAGE_MOCK"
                  size={16}
                />
              }
              initialRating={4}
              readonly={true}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="reviewCount__countSection column small-12 medium-shrink"
    >
      <strong>
        (
        4
        )   
      </strong>
      <a
        className=""
        href="/producers/tweed/reviews"
        onClick={[Function]}
      >
        2
         
        
        reviews
      </a>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="column text-center mt-md"
    >
      <a
        className="button spacious secondary"
        onClick={[Function]}
      >
        <div
          className="row align-middle align-center"
        >
          <div
            className="shrink column"
          >
            View
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
